sdk/devcenter/azure-developer-devcenter/src/main/java/com/azure/developer/devcenter/EnvironmentsClientBuilder.java

```diff
@@ -0,0 +1,338 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.developer.devcenter;
+
+import com.azure.core.annotation.Generated;
+import com.azure.core.annotation.ServiceClientBuilder;
+import com.azure.core.client.traits.ConfigurationTrait;
+import com.azure.core.client.traits.HttpTrait;
+import com.azure.core.client.traits.TokenCredentialTrait;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.HttpPipelinePosition;
+import com.azure.core.http.policy.AddDatePolicy;
+import com.azure.core.http.policy.AddHeadersFromContextPolicy;
+import com.azure.core.http.policy.AddHeadersPolicy;
+import com.azure.core.http.policy.BearerTokenAuthenticationPolicy;
+import com.azure.core.http.policy.CookiePolicy;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.http.policy.HttpLoggingPolicy;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.http.policy.HttpPolicyProviders;
+import com.azure.core.http.policy.RequestIdPolicy;
+import com.azure.core.http.policy.RetryOptions;
+import com.azure.core.http.policy.RetryPolicy;
+import com.azure.core.http.policy.UserAgentPolicy;
+import com.azure.core.util.ClientOptions;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.builder.ClientBuilderUtil;
+import com.azure.core.util.serializer.JacksonAdapter;
+import com.azure.developer.devcenter.implementation.DevCenterClientImpl;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/** A builder for creating a new instance of the EnvironmentsClient type. */
+@ServiceClientBuilder(serviceClients = {EnvironmentsClient.class, EnvironmentsAsyncClient.class})
+public final class EnvironmentsClientBuilder
+        implements HttpTrait<EnvironmentsClientBuilder>,
+                ConfigurationTrait<EnvironmentsClientBuilder>,
+                TokenCredentialTrait<EnvironmentsClientBuilder> {
+    @Generated private static final String SDK_NAME = "name";
+
+    @Generated private static final String SDK_VERSION = "version";
+
+    @Generated private static final String[] DEFAULT_SCOPES = new String[] {"https://devcenter.azure.com/.default"};
+
+    @Generated
+    private final Map<String, String> properties = CoreUtils.getProperties("azure-developer-devcenter.properties");
```

Comment:
For @weidongxu-microsoft, we should make this static as the resource shouldn't change during runtime and doesn't need to be parsed every time constructing a builder.